hal/halx86/legacy/arcmap.c

```diff
@@ -44,57 +50,65 @@ DATA_SEG("INITDATA")
 static
 PNP_MAPPER_DEVICE_ID KeyboardMap[] =
 {
-    { L"XT_83KEY", L"*PNP0300\0" },
-    { L"PCAT_86KEY", L"*PNP0301\0" },
-    { L"PCXT_84KEY", L"*PNP0302\0" },
-    { L"XT_84KEY", L"*PNP0302\0" },
-    { L"101-KEY", L"*PNP0303\0" },
-    { L"OLI_83KEY", L"*PNP0304\0" },
-    { L"ATT_301", L"*PNP0304\0" },
-    { L"OLI_102KEY", L"*PNP0305\0" },
-    { L"OLI_86KEY", L"*PNP0306\0" },
-    { L"OLI_A101_102KEY", L"*PNP0309\0" },
-    { L"ATT_302", L"*PNP030a\0" },
-    { L"PCAT_ENHANCED", L"*PNP030b\0" },
-    { L"PC98_106KEY", L"*nEC1300\0" },
-    { L"PC98_LaptopKEY", L"*nEC1300\0" },
-    { L"PC98_N106KEY", L"*PNP0303\0" },
+    { L"XT_83KEY", L"PNP0300" },
+    { L"PCAT_86KEY", L"PNP0301" },
+    { L"PCXT_84KEY", L"PNP0302" },
+    { L"XT_84KEY", L"PNP0302" },
+    { L"101-KEY", L"PNP0303" },
+    { L"OLI_83KEY", L"PNP0304" },
+    { L"ATT_301", L"PNP0304" },
+    { L"OLI_102KEY", L"PNP0305" },
+    { L"OLI_86KEY", L"PNP0306" },
+    { L"OLI_A101_102KEY", L"PNP0309" },
+    { L"ATT_302", L"PNP030a" },
+    { L"PCAT_ENHANCED", L"PNP030b" },
+    { L"PC98_106KEY", L"nEC1300" },
+    { L"PC98_LaptopKEY", L"nEC1300" },
+    { L"PC98_N106KEY", L"PNP0303" },
     { NULL, NULL }
 };
 
 DATA_SEG("INITDATA")
 static
 PNP_MAPPER_DEVICE_ID PointerMap[] =
 {
-    { L"PS2 MOUSE", L"*PNP0F0E\0" },
-    { L"SERIAL MOUSE", L"*PNP0F0C\0" },
-    { L"MICROSOFT PS2 MOUSE", L"*PNP0F03\0" },
-    { L"LOGITECH PS2 MOUSE", L"*PNP0F12\0" },
-    { L"MICROSOFT INPORT MOUSE", L"*PNP0F02\0" },
-    { L"MICROSOFT SERIAL MOUSE", L"*PNP0F01\0" },
-    { L"MICROSOFT BALLPOINT SERIAL MOUSE", L"*PNP0F09\0" },
-    { L"LOGITECH SERIAL MOUSE", L"*PNP0F08\0" },
-    { L"MICROSOFT BUS MOUSE", L"*PNP0F00\0" },
-    { L"NEC PC-9800 BUS MOUSE", L"*nEC1F00\0" },
+    { L"PS2 MOUSE", L"PNP0F0E" },
+    { L"SERIAL MOUSE", L"PNP0F0C" },
+    { L"MICROSOFT PS2 MOUSE", L"PNP0F03" },
+    { L"LOGITECH PS2 MOUSE", L"PNP0F12" },
+    { L"MICROSOFT INPORT MOUSE", L"PNP0F02" },
+    { L"MICROSOFT SERIAL MOUSE", L"PNP0F01" },
+    { L"MICROSOFT BALLPOINT SERIAL MOUSE", L"PNP0F09" },
+    { L"LOGITECH SERIAL MOUSE", L"PNP0F08" },
+    { L"MICROSOFT BUS MOUSE", L"PNP0F00" },
+    { L"NEC PC-9800 BUS MOUSE", L"nEC1F00" },
     { NULL, NULL }
 };
 
 DATA_SEG("INITDATA")
 static
 PNP_DETECT_IDENTIFIER_MAP PnPMap[] =
 {
-    { L"SerialController", L"*PNP0501\0", NULL, 0 },
+    { L"SerialController", L"PNP0501", NULL, 0 },
     //{ L"KeyboardController", L"*PNP0303\0", NULL, 0 },
     //{ L"PointerController", L"*PNP0F13\0", NULL, 0 },
     { L"KeyboardPeripheral", NULL, KeyboardMap, 0 },
     { L"PointerPeripheral", NULL, PointerMap, 0 },
-    { L"ParallelController", L"*PNP0400\0", NULL, 0 },
-    { L"FloppyDiskPeripheral", L"*PNP0700\0", NULL, 0 },
+    { L"ParallelController", L"PNP0400", NULL, 0 },
+    { L"FloppyDiskPeripheral", L"PNP0700", NULL, 0 },
```

Comment:
(generic comment for all these lists above): Wasn"t there a discussion somewhere (iirc either by Stas'm or disean) who mentioned that these 2nd strings could be MULTI_SZ ?

Reply:
I think he modified the underlying code already, so this is not needed anymore.

Reply:
Yes, the MULTI_SZ is generated in `HalpPdoQueryId`